tests/k8st/utils/utils.py

```diff
@@ -0,0 +1,124 @@
+# Copyright (c) 2018 Tigera, Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import functools
+import logging
+import subprocess
+import time
+
+_log = logging.getLogger(__name__)
+
+
+def start_external_node_with_bgp(name, config):
+    # Setup external node: use privileged mode for setting routes
+    run("docker run -d "
+        "--privileged "
+        "--name %s "
+        "--network kubeadm-dind-net "
+        "mirantis/kubeadm-dind-cluster:tar" % name)
+
+    # It may take a while for the image to download and the container
+    # to start running.
+    while True:
```

Comment:
Might we want to limit this to some amount of time? 

I guess on semaphore we'll eventually time out.

Reply:
Yes, I think relying on the semaphore time out may make sense here.  (Sometimes networking from semaphore is very slow; I'm not sure how to put a reasonable bound on that.)